src/Form/Type/ShopProductsFilterType.php

```diff
@@ -29,6 +61,60 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
             ->add('options', ProductOptionsFilterType::class, ['required' => false, 'label' => false])
             ->add('attributes', ProductAttributesFilterType::class, ['required' => false, 'label' => false])
             ->add('price', PriceFilterType::class, ['required' => false, 'label' => false])
-        ;
+            ->setMethod('GET');
+
+        $builder->addEventListener(FormEvents::PRE_SUBMIT, [$this, 'resolveFacets']);
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'csrf_protection' => false,
+        ]);
+    }
+
+    public function resolveFacets(FormEvent $event): void
```

Comment:
what you think to move this logick to serparate object e.g. "ProductsFilterFacetResolver" ?